tests/datasource/fluent/test_spark_datasource.py

```diff
@@ -53,3 +72,48 @@ def test_dataframe_asset(
             for asset in datasource.assets.values()
         ]
     )
+
+
+@pytest.mark.xfail(
+    strict=True,
+    reason="this will fail until we add batch_metadata to Spark add_*_asset methods",
+)
```

Comment:
@billdirks your PR for Spark `batch_metadata`, should cause you to remove this xfail.

Reply:
This actually still fails because the `DataFrameAsset` has not implemented `build_batch_request`. Would you like me to implement this here or can I do it in a follow up PR?